vizro-core/src/vizro/models/_components/form/range_slider.py

```diff
@@ -40,7 +40,7 @@ class RangeSlider(VizroBaseModel):
     min: Optional[float] = Field(None, description="Start value for slider.")
     max: Optional[float] = Field(None, description="End value for slider.")
     step: Optional[float] = Field(None, description="Step-size for marks on slider.")
-    marks: Optional[Dict[float, str]] = Field({}, description="Marks to be displayed on slider.")
+    marks: Optional[Dict[int, Union[str, Dict[str, str]]]] = Field({}, description="Marks to be displayed on slider.")
```

Comment:
I am not sure I understand this change. Did I miss something?

Reply:
This change is not connected to this bug fix, but I included it in this PR as it is fix on the sliders. This 'bug' was discovered yesterday by Peter when he tried to show marks on a RangeSlider. 
Basically, our previous type for `marks` which was `Optional[Dict[float, str]]` was disabling marks to be shown on the sliders as dcc sliders dictionary keys must be integers for the labels to be displayed. 

Additionally, dcc sliders have a handy way to styling individual marks, which was not supported by us so far, and that is achieved by providing marks values as a dictionary, something like below. :

To display marks on sliders I had to change float -> str, and to have this ability to style individual marks, i changed str -> Union[str, Dict[str, Any]]. 
`marks={1: {'label': "Jan 23", 'style': {'fontSize': '8px', "transform": "rotate(40deg)"}}}`

<img width="763" alt="Screenshot 2024-01-16 at 18 10 54" src="https://github.com/mckinsey/vizro/assets/108531476/fc340927-2784-4551-85b8-039a94712d05">